hybris/tests/test_hwcomposer.cpp

```diff
@@ -119,7 +119,7 @@ void HWComposer::present(HWComposerNativeWindowBuffer *buffer)
 	assert(err == 0);
 
 	err = hwcdevice->set(hwcdevice, HWC_NUM_DISPLAY_TYPES, mlist);
-	assert(err == 0);
+	//assert(err == 0);
```

Comment:
From code (4.2.2 & 4.4.2): basically because HWC in rockchip devices only supports HWC_DISPLAY_PRIMARY (=0), all other display types err; so setting all HWC_NUM_DISPLAY_TYPES (=3) by hwcdevice->set(hwcdevice, HWC_NUM_DISPLAY_TYPES, mlist) fails. And as a hack: hwcdevice->set(hwcdevice, 1, mlist) succeeds...

However, more importantly, comparing with android: in surfaceflinger (also in AOSP) this doesnt actually matter as the return value of mHwc->set() in https://android.googlesource.com/platform/frameworks/native/+/master/services/surfaceflinger/DisplayHardware/HWComposer.cpp#782 isnt used when called by https://android.googlesource.com/platform/frameworks/native/+/master/services/surfaceflinger/SurfaceFlinger.cpp#1255
However, here in test_hwcomposer.cpp the error is asserted and program stops.